src/main/java/core/basesyntax/operation/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        int fruitsInStorage = Storage.storage.get(fruitTransaction.getFruit());
+        int updateQuantity = fruitsInStorage + fruitTransaction.getQuantity();
+        Storage.storage.put(fruitTransaction.getFruit(), updateQuantity);
```

Comment:
While updating the storage, you should check if the `fruitTransaction.getQuantity()` is a non-negative number to prevent incorrect storage updates.